src/components/accordion-disclosure/index.tsx

```diff
@@ -11,18 +12,48 @@ import {
 	useAccordionDisclosureContext,
 	AccordionDisclosureContext,
 } from './accordion-disclosure-context'
+import useHover from 'hooks/use-hover'
 
 const AccordionDisclosure = ({
 	children,
 	description,
 	initialOpen,
 	title,
+	groupData,
+	activatorHeadingLevel,
 }: AccordionDisclosureProps) => {
+	/**
+	 * Note: useHover is used here to allow us to change the styling
+	 * of the outer `root` container component when the `<button />`
+	 * within, which is the functional interactive element, is hovered.
+	 *
+	 * Since we're changing styles on an ancestor node of the `<button />`
+	 * whose `:hover` state we want to track, this is not as feasible
+	 * with pure CSS.
+	 */
+	const [hoverRef, isHovered] = useHover<HTMLButtonElement>()
```

Comment:
I think we should go for a CSS approach here before adding a `useHover` hook that's purely for changing styles and not adding component functionality. What we want to do should be possible with some rearrangement of the HTML elements. Rather than applying some styles to the parent of `<button>`, we could add a sibling element next to it that we apply styles to when `<button>` is hovered. It would look like a container, but really be an absolutely positioned element that's behind `<button>`.

Reply:
I think this probably relates to the "heading" wrapper and logic. My understanding so far has been that we'll want to add the "heading" wrapper to match [the Accordion pattern](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/), which makes the sibling element selector approach less feasible. It would also seem to break the pattern, which mentions `button` should be the only child of the heading element:

> The button element is the only element inside the heading element. That is, if there are other visually persistent elements, they are not included inside the heading element. 

That being said, if we do feel we need to drop the `heading` wrapper, then I can see how using sibling selectors and an absolutely positioned element could become more feasible 👍

Reply:
Taking another look at this after removing the `heading` stuff. However, I'm still seeing some issues - namely that `validate-disclosure-children` expects exactly `2` children, so adding a absolutely positioned sibling element to `button` would require refactoring the validation function. As well, it would break the `.activator[aria-expanded='false'] + .contentContainer ` in `disclosure.module.css`.

I think the `useHover` hook here is a reliable way to achieve the same thing, while leaving our accessibility checks intact. Does `useHover` seem like an acceptable solution given those constraints?

Reply:
I think there are some things we can try, and I'm seeing less value now in `validateDisclosureChildren`, but we can defer.

Reply:
Sounds good - will defer for now, very open to trying stuff later 💯